field_audit/models.py

```diff
@@ -568,13 +573,44 @@ def delete(self, *, audit_action=AuditAction.RAISE):
         return value
 
     @validate_audit_action
-    def update(self, *args, audit_action=AuditAction.RAISE, **kw):
+    def update(self, audit_action=AuditAction.RAISE, **kw):
         if audit_action is AuditAction.IGNORE:
-            return super().update(*args, **kw)
-        else:
-            raise NotImplementedError(
-                "Change auditing is not implemented for update()."
-            )
+            return super().update(**kw)
+        assert audit_action is AuditAction.AUDIT, audit_action
+
+        fields_to_update = set(kw.keys())
+        audited_fields = set(
+            getattr(self.model, AuditEvent.ATTACH_FIELD_NAMES_AT)
+        )
+        fields_to_audit = fields_to_update & audited_fields
+        if not fields_to_audit:
+            # no audited fields are changing
+            return super().update(**kw)
+
+        # since we fetch objects from this queryset, save old values, update,
+        # and then fetch the objects again, we need to lock until this is done
+        with transaction.atomic():
```

Comment:
Offlined with @gherceg about this. I gave misleading feedback previously. A transaction doesn't provide any database record locking, and therefore doesn't provide any "previous value" guarantee here.

Reply:
c40f738bc255f230de35ad60fdab9b0447f7d6b7